src/ZoneServer/World/Maps/Map.cs

```diff
@@ -190,8 +191,20 @@ private void Disappearances()
 			lock (_monsters)
 				toDisappear = _monsters.Values.Where(a => a.DisappearTime < now).ToList();
 
-			foreach (var monster in toDisappear)
+			foreach (var monster in toDisappear) {
+
+				// If this is a blue orb summon, inform the summoner that it expired
+				if (monster is Mob mob) {
+					if (mob.MonsterType != MonsterType.Mob && mob.Components.Get<AiComponent>()?.Script.GetMaster() != null && mob.Components.Get<AiComponent>()?.Script.GetMaster() is Character master)
```

Comment:
If you're going to call multiple methods on one object, such as an AI script or a character, you should create a reference to shorten the code a little. This makes it easier to read and improves performance.

```cs
if (monster is Mob mob)
{
    var master = mob.Components.Get<AiComponent>()?.Script.GetMaster();

    if (mob.MonsterType != MonsterType.Mob && master is Character masterCharacter)
```

Reply:
I originally did it that way because I wasn't sure if it was safe against null.  It does appear to be, though (I just ran a test on it), so I've simplified it.